packages/e2e-tests/test/e2e-current-op.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { expect } from 'chai';
+import {
+  skipIfApiStrict,
+  startSharedTestServer,
+} from '../../../testing/integration-testing-hooks';
+import type { TestShell } from './test-shell';
+
+describe('e2e currentOp', function () {
+  skipIfApiStrict();
+
+  const testServer = startSharedTestServer();
+
+  context('with 2 shells', function () {
+    let helperShell: TestShell;
+    let currentOpShell: TestShell;
+
+    const OPERATION_TIME = 1_000;
+    const CURRENT_OP_WAIT_TIME = 100;
+    this.timeout(OPERATION_TIME * 5);
+
+    beforeEach(async function () {
+      helperShell = this.startTestShell({
+        args: [await testServer.connectionString()],
+      });
+      currentOpShell = this.startTestShell({
+        args: [await testServer.connectionString()],
+      });
+      await helperShell.waitForPrompt();
+      await currentOpShell.waitForPrompt();
+
+      // Insert a dummy object so find commands will actually run with the delay.
+      await helperShell.executeLine('db.coll.insertOne({})');
+    });
+
+    it('should return the correct operation', async function () {
+      const regexOperation = helperShell.executeLine(
+        `db.coll.find({$where: function() { sleep(${OPERATION_TIME}) }}).projection({test: 1})`
+      );
+      helperShell.assertNoErrors();
+      await currentOpShell.executeLine(`sleep(${CURRENT_OP_WAIT_TIME})`);
+      let currentOpCall = await currentOpShell.executeLine(`db.currentOp()`);
+
+      currentOpShell.assertNoErrors();
+      expect(currentOpCall).to.include("find: 'coll'");
+      expect(currentOpCall).to.include(
+        `filter: { '$where': Code('function() { sleep(${OPERATION_TIME}) }') }`
+      );
+      expect(currentOpCall).to.include('projection: { test: 1 }');
+
+      await regexOperation;
+
+      currentOpCall = await currentOpShell.executeLine(`db.currentOp()`);
+
+      currentOpShell.assertNoErrors();
+      expect(currentOpCall).not.to.include("find: 'coll'");
+      expect(currentOpCall).not.to.include(
+        `filter: { '$where': Code('function() { sleep(${OPERATION_TIME}) }') }`
+      );
+      expect(currentOpCall).not.to.include('projection: { test: 1 }');
```

Comment:
I wonder if you could make the test faster and still test what you want, by simply killing the test shell instead of awaiting the `regexOperation` and asserting the current op gets reset?

Reply:
I agree it'd still be the test I want for the purposes of changes in this PR; I thought this is nice in terms of it provides a full e2e usecase in terms of currentOp also working after the operation ended but that's not responsibility of this change

Reply:
@kraenhansen  Ah right, looking the exact code again. I actually made 2 tests, since this is also introducing e2e test to currentOp in general:
1. **one "generic"  e2e (unrelated to the changes):** which just makes sure the currentOp works as expected; I think here it's nice to await the command (which confused me because I still call it `regexOperation`, it actually has no `regex` so I'll rename this) and make sure that running currentOp again updates accordingly. I think it's nice to have it just for general currentOp testing.
2. **the test related to these changes** in which case I no longer care about the operation and it doesn't get awaited as you describe.